utils/uevent/events.go

```diff
@@ -0,0 +1,58 @@
+package uevent
+
+import (
+	"encoding/json"
+	"fmt"
+	"slices"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/gogoproto/proto"
+	"golang.org/x/exp/maps"
+)
+
+// EmitTypedEvent takes a typed event and emits it
+func EmitTypedEvent(ctx sdk.Context, tev proto.Message) error {
```

Comment:
You can just call it `EmitTyped` since it's in uevent

Reply:
this follows the original naming in the sdk

Reply:
are you sure? this will mean renaming it all over the hub repo, and anywhere else where this method is used.